### PR TITLE
Final touch ups to refactored code before merging to main.

### PR DESCRIPTION
**Pull request by averma-canonical**
Addressing review comments shared by Rob.

### CODE REVIEW

File: spark_client/domain.py [hunk 1]
```diff
@@ -207,7 +207,7 @@ def kubectl_cmd(self) -> str:
 
     @property
     def scala_history_file(self):
-        return f"{self.home_folder}/.scala_history"
+        return f"{self.environ['SNAP_USER_DATA']}/.scala_history"
```
Comment (deusebio) [inline spark_client/domain.py:210]: maybe we can just have another property snap_user_data

**Review by deusebio — APPROVED** [review]
All good! Just a minor comment

File: spark_client/utils.py [hunk 2]
```diff
@@ -182,3 +182,59 @@ def environ(*remove, **update):
 
 def listify(value: Any) -> List[str]:
     return [str(v) for v in value] if isinstance(value, list) else [str(value)]
+
+
+def parse_arguments_with(parsers=[], namespace=None):
```
Comment (grobbie) [inline spark_client/utils.py:187]: please provide some code commentary to the methods in this file

**Review by grobbie — CHANGES_REQUESTED** [review]
thanks, just one action for resolution please

**Review by grobbie — APPROVED** [review]
thanks, looks good

**Pull request merged**